### PR TITLE
Whitelist: Implement whitelist tokens functionality in Smart Contracts

### PR DESCRIPTION
**Pull request by ldgaetano**

### CODE REVIEW

File: src/main/resources/contracts/Lilium/BoxGuardScripts/IssuerContract.es [hunk 1]
```diff
@@ -1,55 +1,85 @@
...
-    val proof = getVar[Coll[Byte]](0).get
+    val validNFTIssuanceBox: Boolean = {
 
+      val validToken: Boolean = {
 
-    val buyerPropBytes: SigmaProp = SELF.R9[(SigmaProp, Long)].get._1.propBytes
-    val index: Long = SELF.R9[(SigmaProp, Long)].get._2
+        allOf(Coll(
+          (OUTPUTS(0).tokens(0)._1 == SELF.id),
+          (OUTPUTS(0).tokens.size == 1L)
```
Comment (ldgaetano) [inline src/main/resources/contracts/Lilium/BoxGuardScripts/IssuerContract.es]: Oh I think this line should be `(OUTPUTS(0).tokens(0)._2 == 1L)`

Comment (lucagdangelo) [reply src/main/resources/contracts/Lilium/BoxGuardScripts/IssuerContract.es]: Ok I fixed it in a new commit

Comment (ldgaetano) [reply src/main/resources/contracts/Lilium/BoxGuardScripts/IssuerContract.es]: > Oh I think this line should be `(OUTPUTS(0).tokens(0)._2 == 1L)`

Ok I fixed it in a new commit

**Comment by mgpai22** [comment]
@lucagdangelo please review changes

- Proxy Contract still needs a condition which ensures all tokens of SELF are transferred to buyer

**Pull request merged**